app/src/test/java/at/aau/serg/network/AuthenticationTest.kt

```diff
@@ -0,0 +1,124 @@
+package at.aau.serg.network
+
+import android.content.ContextWrapper
+import android.content.SharedPreferences
+import at.aau.serg.logic.Authentication
+import at.aau.serg.models.User
+import com.google.gson.Gson
+import io.mockk.Called
+import io.mockk.Runs
+import io.mockk.clearAllMocks
+import io.mockk.every
+import io.mockk.just
+import io.mockk.mockk
+import io.mockk.verify
+import okhttp3.Callback
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNull
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+
+class AuthenticationTest {
+
+    private lateinit var authentication: Authentication
+    private lateinit var httpClient: HttpClient
+    private lateinit var callback: Callback
+    @BeforeEach
+    fun setUp() {
+        authentication = Authentication()
+        httpClient = mockk()
+        callback = mockk(relaxed = true)
+    }
+
+    @AfterEach
+    fun tearDown() {
+        clearAllMocks()
+    }
+
+    @Test
+    fun `registerUser with valid credentials`() {
+        val username = "testUser"
+        val password = "testPassword"
+        val userToRegister = User(username, password)
+        val expectedRequestBody = Gson().toJson(userToRegister)
+
+        every { httpClient.post(any(), any(), any(), any()) } just Runs
```

Comment:
The problem with this test is that you create an `HTTPClient` mock but never use it since you create a new `HTTPClient` for every request in `Authentication.kt`. You can fix this by passing an `HTTPClient` to the `Authentication` class and using it inside the functions (it also helps since we don't have to create a new `HTTPClient` for each call).

We could also consider making the HTTPClient a singleton, but it is not required right now.

Reply:
ty, it works now